dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/datasetreport/impl/DefaultDataSetReportService.java

```diff
@@ -316,13 +318,56 @@ else if ( formType.isSection() )
             tmpDataSet.getSections().add( section );
         }
 
-        return getSectionDataSetReport( tmpDataSet, period, unit, filters, selectedUnitOnly );
+        return getSectionDataSetReport( tmpDataSet, periods, unit, filters, selectedUnitOnly );
     }
 
     // -------------------------------------------------------------------------
     // Supportive methods
     // -------------------------------------------------------------------------
 
+    private String formatPeriods( List<Period> periods, I18nFormat format )
+    {
+        Collections.sort( periods );
+
+        List<String> formattedPeriods = periods.stream()
+            .map( p -> format.formatPeriod( p ) )
+            .collect( Collectors.toList() );
+
+        return Joiner.on( ", " ).join( formattedPeriods );
+    }
+
+    /**
+     * Returns the sum of values for the list of periods, but returns null if
+     * all the values are null.
+     *
+     * @param dataElement the data element.
+     * @param periods the periods.
+     * @param unit the organisation unit.
+     * @param optionCombo the category option combo.
+     * @return the combined value.
+     */
+    private Double getSelectedUnitValue( DataElement dataElement, List<Period> periods, OrganisationUnit unit,
```

Comment:
You could also do this as stream:
* stream periods
* map to `DataValue`
* filter `null`
* map to `getValue`
* filter `null`
* collect to list to check if empty (return null)
* `mapToDouble` on list and `sum()`.

Reply:
Also done.